### PR TITLE
Widgets v2

### PR DESCRIPTION
**Pull request by rummik**
**Currently not ready for merge, but ready for discussion.**

- [ ] `I/We give` button
- [ ] `I/We receive` button
- [ ] Gulp
  - [x] Gulpfile.js
  - [x] JSHint (v1/v2)
  - [ ] Unit tests
    - [ ] Split v2 image generation code into a local lib for unit tests
    - [ ] Image diff (v2)
    - [ ] Protractor tests (v1/v2)
      - [x] Export `app` in `app.js` for use with Gulp task
  - [ ] v1 build tasks
    - [x] Uglify.js
    - [x] HTMLmin
- [ ] Reference gratipay-badge, Shields.io, and other related projects in README.md

### CODE REVIEW

**Comment by whit537** [comment]
[IRC](https://botbot.me/freenode/gratipay/2015-01-27/?msg=30597592&page=2)

**Comment by whit537** [comment]
Okay! Next thing we need here is a todo list, ya? [IRC](https://botbot.me/freenode/gratipay/2015-01-27/?msg=30597583&page=2)

**Comment by rummik** [comment]
@whit537 Added a todo list, will be checking boxes over the next few days

**Comment by techtonik** [comment]
@rummik, how about splitting this for incremental review? Gulp stuff seems worthy and orthogonal to widget versioning.

**Comment by whit537** [comment]
Good idea, @techtonik! :-)

File: README.md [hunk 1]
```diff
@@ -1,130 +1,85 @@
...
 ```
 $ git clone git@github.com:gratipay/grtp.co.git
 $ cd grtp.co
-$ make run
+$ make test run
 ```
 
-Then visit http://localhost:9537/test/
-
+The widget server should then be listening on http://localhost:5000/.
```
Comment (techtonik) [inline README.md]: Why the change?

Comment (rummik) [inline README.md]: Standard test port for Node.js projects.  Just makes sense to use the standard, since, iirc, /test/ isn't really what anyone would expect it to be.
